lib/one_model.ex

```diff
@@ -243,6 +249,11 @@ defmodule OneModel do
       defp do_select(query, key) when is_atom(key), do: select(query, [b], field(b, ^key))
       defp do_select(query, keys), do: select(query, ^keys)
```

Comment:
Your other option to provide this functionality would be to drop this line and put the `do` of line 255 here. I don't believe that anything is currently using `select: list`, partly because the return value is pretty ugly, and you could change that operation to return a map instead. Saves us an option, and eliminates an operation that is unlikely to ever be used, especially if this one is available.

In other words, don't implement a new option, just put the sensible part of this new operation in place of the rather ugly solution that is almost certainly not use for select: list.

Reply:
I did that in my original implementation of support the select option. Steve rejected it because it's inconsistent with Ecto's API. Passing the select option  a list of keys in Ecto  returns a schema. But I think my suggestion should be a good way around that.

Reply:
Our goal  is, and should be, to be consistent with Ecto for internal use. 

Ecto has 2 options for select.
* select: [:one, :two] - returns the struct with only those 2 keys fetched. all other fields are the schema's defaults
* select: map(binding, [:one, :two]) - returns a map with just those 2 fields.

This is the behaviour I want in our app. So it's just a matter of the api that translates into those 2 options. 

The simplest and most concise approach is to have OneModel allow a new key for selecting a map. Which is what I've implemented.

See my reply on comment below for results of my conversation with Paul on this topic.